common.lic

```diff
@@ -391,42 +391,50 @@ module DRC
     g
   end
 
-  def play_song?(settings, song_list, worn = true, skip_clean = false)
+  def play_song?(settings, song_list, worn = true, skip_clean = false, climbing = false)
     instrument = worn ? settings.worn_instrument : settings.instrument
     UserVars.song = song_list.first.first unless UserVars.song
+    UserVars.climbing_song = @song_list.first.first unless UserVars.climbing_song || !climbing
+    song_to_play = climbing ? UserVars.climbing_song : UserVars.song
     fput('release ecry') if DRSpells.active_spells["Eillie's Cry"].to_i > 0
-    result = bput("play #{UserVars.song}", 'dirtiness may affect your performance', 'slightest hint of difficulty', 'fumble slightly', /Your .+ is submerged in the water/, 'You begin a', 'You struggle to begin', 'You\'re already playing a song', 'You effortlessly begin', 'You begin some', 'You cannot play', 'Play on what instrument', 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play', 'You should stop practicing')
+    result = bput("play #{song_to_play}", 'dirtiness may affect your performance', 'slightest hint of difficulty', 'fumble slightly', /Your .+ is submerged in the water/, 'You begin a', 'You struggle to begin', 'You\'re already playing a song', 'You effortlessly begin', 'You begin some', 'You cannot play', 'Play on what instrument', 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play', 'You should stop practicing')
     case result
     when 'Play on what instrument'
       snapshot = "#{right_hand}#{left_hand}"
       fput("get #{instrument}")
       return false if snapshot == "#{right_hand}#{left_hand}"
       fput("wear #{instrument}") if worn
-      play_song?(settings, song_list, worn, skip_clean)
+      play_song?(settings, song_list, worn, skip_clean, climbing)
     when 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play', 'You should stop practicing'
       false
     when 'You\'re already playing a song'
       fput('stop play')
-      play_song?(settings, song_list, worn, skip_clean)
+      play_song?(settings, song_list, worn, skip_clean, climbing)
     when 'You cannot play'
       wait_for_script_to_complete('safe-room')
     when 'dirtiness may affect your performance'
       return true if DRSkill.getrank('Performance') < 20
       return true if skip_clean
       return true unless clean_instrument(settings, worn)
-      play_song?(settings, song_list, worn, skip_clean)
+      play_song?(settings, song_list, worn, skip_clean, climbing)
     when 'slightest hint of difficulty', 'fumble slightly'
       true
     when 'You begin a', 'You effortlessly begin', 'You begin some'
-      return true if UserVars.song == 'concerto masterful'
+      return true if song_to_play == 'concerto masterful'
```

Comment:
💡 Should this check something like `song_list.last.last` instead of a hardcoded string?

Reply:
Uh, perhaps.  base-perform is a hash of arrays, and for some mindblowing reason this Ruby allows hash.first but not hash.last.   So it'll be song_list.to_a.last.last, which is slightly weirder.

Reply:
Not a big deal, you don't need to change this one.

Reply:
I did check that song_list.to_a.last.last returns concerto masterful, so I just did it anyway.

Reply:
That's fine too!